### PR TITLE
Add RSM Dev Container Templates to collection-index.yml

### PR DESCRIPTION
**Pull request by myty**
This pull request adds the RSM Dev Container Templates to the collection-index.yml file.

### CODE REVIEW

File: _data/collection-index.yml [hunk 1]
```diff
@@ -657,4 +657,9 @@
...
   contact: https://github.com/shibayan/devcontainers/issues
   repository: https://github.com/shibayan/devcontainers
-  ociReference: ghcr.io/shibayan/devcontainers
\ No newline at end of file
+  ociReference: ghcr.io/shibayan/devcontainers
+- name: Dev Container Templates by RSM HCD Engineering
+  maintainer: RSM HCD Engineering
+  contact: https://github.com/rsm-hcd
+  repository: https://github.com/rsm-hcd/devcontainer-templates
+  ociReference: ghcr.io/rsm-hcd/devcontainer-templates
```
Comment (samruddhikhandale) [inline _data/collection-index.yml:665]: Hi 👋

Can you help change the visibility of this metadata package (and respective Template packages) to public? The [index](https://containers.dev/templates) won't be able to crawl for the Templates in `rsm-hcd/devcontainer-templates` if it's private.

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Thanks for the addition, left one comment.

File: _data/collection-index.yml (hunk 1 above)
Comment (myty) [reply _data/collection-index.yml:665]: Just changed all of the packages to public.

**Review by samruddhikhandale — APPROVED** [review]
Looks good, thanks!

**Pull request merged**